data-service/src/main/java/uk/gov/gchq/palisade/service/data/model/AuditableDataRequest.java

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright 2018-2021 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package uk.gov.gchq.palisade.service.data.model;
+
+
+import uk.gov.gchq.palisade.Generated;
+
+import java.util.Objects;
+import java.util.StringJoiner;
+
+/**
+ * The class contains the authorised access to the resource.  This class is a container for the {@link DataRequest}
+ * {@link DataResponse} and {@link AuditErrorMessage} during stream processing.  The container will hold the
+ * {@code AuditErrorMessage} when there has been an error in the process.
+ */
+public final class AuditableDataRequest {
```

Comment:
This class has seen a few name changes and none have quite worked.  None have quite managed.  The root of the problem is there are two steps in the getting the client's response.  The first  call is to authoriseRequest method which will retrieve the list of resources from the database.  This then input to the next step which is to get the actual resources.  I think that if I change DataResponse to AuthorisedData and with it the Auditable container class to be AuditableAuthorisedData it would hopefully be clearer what was meant.